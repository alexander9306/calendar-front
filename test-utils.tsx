// This is a prepare dependecy test - that's why this rule is disable
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render, RenderResult } from "@testing-library/react";
import { ApolloProvider } from "@apollo/client";
import { setupServer } from "msw/node";
import userEvent from "@testing-library/user-event";
import client from "./src/apollo-client/client";
import requestHandlers from "./request-handlers";

const AllTheProviders: React.FunctionComponent = ({ children }) => {
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type CustomRender = (ui: JSX.Element, options?: any) => RenderResult;

const customRender: CustomRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };

const server = setupServer();

beforeAll(() => {
  server.listen({
    onUnhandledRequest: "error",
  });
  client.resetStore();
});
afterEach(() => {
  server.resetHandlers();
});

afterAll(() => {
  server.close();
  client.clearStore();
});

export { server, userEvent };
