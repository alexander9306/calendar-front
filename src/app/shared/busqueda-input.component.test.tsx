import React from "react";
import { create } from "react-test-renderer";
import { render, screen, userEvent, waitFor } from "test-utils";
import BusquedaInput from "./busqueda-input.component";

describe("Busqueda input Component", () => {
  test("Verify component renders correctly", () => {
    const tree = create(
      <BusquedaInput
        placeholder="Buscar ordenes..."
        name="orden"
        onSubmit={() => ({})}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Verify component readability", () => {
    test("Verify placeholder text", () => {
      render(
        <BusquedaInput
          placeholder="Buscar ordenes..."
          name="orden"
          onSubmit={() => ({})}
        />
      );
      expect(
        screen.getByPlaceholderText("Buscar ordenes...")
      ).toBeInTheDocument();
    });

    test("Verify parameters are passed with the same name to the onSubmit function after a submit", async () => {
      const onSubmit = jest.fn();
      render(
        <BusquedaInput
          placeholder="Buscar ordenes..."
          name="orden"
          onSubmit={onSubmit}
        />
      );

      const input = await screen.findByPlaceholderText("Buscar ordenes...");
      userEvent.type(input, "112");

      const submitBtn = await screen.findByText("Buscar");
      userEvent.click(submitBtn);

      await waitFor(() => {
        expect(onSubmit).toHaveBeenCalledTimes(1);
      });

      expect(onSubmit.mock.calls[0][0]).toEqual({
        orden: "112",
      });
    });

    test("Verify parameters are passed with the same name to the onSubmit function after a submit but and with the right value", async () => {
      const onSubmit = jest.fn();
      render(
        <BusquedaInput
          placeholder="Buscar ordenes..."
          name="orden"
          onSubmit={onSubmit}
        />
      );

      const input = await screen.findByPlaceholderText("Buscar ordenes...");
      userEvent.type(input, "112");

      const submitBtn = await screen.findByText("Buscar");
      userEvent.click(submitBtn);

      await waitFor(() => {
        expect(onSubmit.mock.calls[0][0]).not.toEqual({
          orden: "110",
        });
      });
    });
  });
});
