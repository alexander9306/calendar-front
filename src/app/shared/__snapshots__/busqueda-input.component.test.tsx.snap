// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Busqueda input Component Verify component renders correctly 1`] = `
<form
  className="form-horizontal "
  method="post"
  onSubmit={[Function]}
  role="search"
>
  <div
    className="row form-group"
    css={
      Object {
        "map": undefined,
        "name": "15bfgzn",
        "next": undefined,
        "styles": "
        @media screen and (min-width: 768px) {
          float: right;
        }
        ",
        "toString": [Function],
      }
    }
  >
    <div
      className="col-md-12"
    >
      <div
        className="input-group input-group-md"
      >
        <input
          className="form-control"
          name="orden"
          placeholder="Buscar ordenes..."
          type="search"
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="submit"
          >
            Buscar
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
